compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -349,12 +349,15 @@ class ClassfileParser(
       val tag = sig(index); index += 1
       (tag: @switch) match {
         case 'L' =>
-          def processInner(tp: Type): Type = tp match {
-            case tp: TypeRef if !tp.symbol.owner.is(Flags.ModuleClass) =>
-              TypeRef(processInner(tp.prefix.widen), tp.symbol.asType)
-            case _ =>
-              tp
-          }
+          /** Inner class references of A.this.B into A#B, do the same transformation recursively on `A` */
```

Comment:
That comment is no longer accurate when the input is a symbol, I suggest:
```suggestion
          /** A type representation where inner classes become `A#B` instead of `A.this.B` (like with `typeRef`) */
```